Assets/Scripts/Traffic System/CarCollision.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarCollision : MonoBehaviour
+{
+    [SerializeField] GraphMovement car;
+    float carSpeed;
+    [SerializeField] float visionAngle; //radians
+    
+    private void Start() {
+        carSpeed = car.speed;
+        }
+    private void OnTriggerEnter(Collider other) {
+        Vector3 agentToVVertex = car.transform.position - other.gameObject.transform.position;
+        agentToVVertex.Normalize();
+        if (Vector3.Dot(agentToVVertex, car.transform.forward) > Mathf.Cos(visionAngle)) {
+            if (other.gameObject.CompareTag("Car")) {
+                float otherSpeed = other.gameObject.GetComponent<GraphMovement>().speed;
+                if (otherSpeed > 0) {
+                    car.speed = otherSpeed - 2;
+                    }
+                else {
+                    car.speed = 0;
+                    }
+                    
+
+
+                }
+            else if (other.gameObject.CompareTag("Person")) {
+                car.speed = 0;
+                }
+            
+            }
+        }
+    private void OnTriggerExit(Collider other) {
+        if (other.gameObject.CompareTag("Car") || other.gameObject.CompareTag("Person")) {
```

Comment:
may want to make tags serializable later depending on what we want to do with this--but for now this is fine the way it is